R/export_cdisc.R

```diff
@@ -206,3 +168,132 @@ find_common_prefix <- function(strings) {
 
   substr(strings[[1]], 0, which(mismatch)[1] - 1)
 }
+
+#' Helper function to extract SUBJID from data
+#'
+#' This function extracts the `SUBJID` from the provided dataset. If `SUBJID` is not available,
+#' it attempts to derive it from `USUBJID` by removing the `STUDYID` prefix or the common prefix
+#' shared by all `USUBJID` values.
+#'
+#' @details
+#' The function first checks if `SUBJID` exists in the dataset. If not, it derives `SUBJID` from
+#' `USUBJID` by:
+#' \itemize{
+#'   \item Removing the `STUDYID` prefix if it exists.
+#'   \item Removing the longest common prefix shared by all `USUBJID` values.
+#' }
+#' If neither `SUBJID` nor `USUBJID` is available, the function returns `NA`.
+#'
+#' @param data A data frame containing `SUBJID`, `USUBJID`, and optionally `STUDYID`.
+#'
+#' @returns A vector of `SUBJID` values.
+#'
+#' @examples
+#' data <- data.frame(
+#'   STUDYID = c("STUDY1", "STUDY1"),
+#'   USUBJID = c("STUDY1-001", "STUDY1-002")
+#' )
+#' get_subjid(data)
+#'
+#' @noRd
+#' @keywords internal
+get_subjid <- function(data) {
+  if ("SUBJID" %in% names(data)) {
+    data$SUBJID
+  } else if ("USUBJID" %in% names(data)) {
+    if ("STUDYID" %in% names(data)) {
+      stringr::str_remove(as.character(data$USUBJID), paste0(as.character(data$STUDYID), "\\W?"))
+    } else {
+      gsub(find_common_prefix(data$USUBJID), "", data$USUBJID)
+    }
+  } else {
+    NA
+  }
+}
+
+
+CDISC_COLS <- list(
+  ADPC = c(
+    "STUDYID",
+    "SUBJID",
+    "USUBJID",
+    "SITEID",
+    "VISITNUM",
+    "VISIT",
+    "AVISIT",
+    "AVISITN",
+    "PCSTRESC",
+    "PCSTRESN",
+    "PCSTRESU",
+    "PCORRES",
+    "PCORRESU",
+    "PCTPT",
+    "PCTPTNUM",
+    "ATPT",
+    "ATPTN",
+    "PCSTRESC",
+    "PCSTRESN",
+    "PCSTRESU",
+    "AVAL",
+    "ANL01FL"
+  ),
+
+  ADPP = c(
+    "STUDYID",
+    "USUBJID",
+    "PPSEQ",
+    "PPGRPID",
+    "PPSPID",
+    "PARAMCD",
+    "PARAM",
+    "PPCAT",
+    "PPSCAT",
+    "PPREASND",
+    "PPSPEC",
+    "PPDTC",
+    "PPSTINT",
+    "PPENINT",
+    "SUBJID",
+    "SITEID",
+    "SEX",
+    "RACE",
+    "ACTARM",
+    "AAGE",
+    "AAGEU",
```

Comment:
```suggestion
    "AGE",
    "AGEU",
```

spelling mistake? or is it meant to be like that